### PR TITLE
.NET Standard 2.0 conversion

### PR DESCRIPTION
**Pull request by Nevett**
Converted with minimal changes to the code itself.
- Produces a single .NET Standard 2.0 binary and NuGet package supported by .NET Framework 4.6.1 and Core 2.0
- Small bump to the `Newtonsoft.Json` minimum version
- Changed example project to .NET Core
- Set up a GitHub Action to run the tests
- Template files removed thanks to nicer csproj features.
- Things have moved about a bit in the build, mostly thanks to dotnet CLIs combined build and package command, but the results should be equivalent.
- Brought in icon file per newer Nuget spec.
- Updated styling in line with modern StyleCop analyzers, with some selected overrides

To develop locally, install the latest [dotnet SDK](https://dotnet.microsoft.com/download) and try
```
dotnet test
dotnet run --project src/Cronofy.Example
```

### CODE REVIEW

**Comment by GreenAirplane** [comment]
Hi @Nevett, any idea on when this pull request can be merged? Let me know if you need any help.

**Comment by Nevett** [comment]
> Hi @Nevett, any idea on when this pull request can be merged? Let me know if you need any help.

Hey, we are planning to get this merged in still - thanks for reaching out so we can see the demand!

In the short term if you're blocked waiting for this you can actually reference https://www.nuget.org/packages/CronofyCore which is a build of this branch under a different package name which some other customers have been successfully using.

File: CHANGELOG.md [hunk 1]
```diff
@@ -267,7 +272,8 @@
 [#76]: https://github.com/cronofy/cronofy-csharp/pull/76
 [#77]: https://github.com/cronofy/cronofy-csharp/pull/77
 [#80]: https://github.com/cronofy/cronofy-csharp/pull/80
-[#82]: https://github.com/cronofy/cronofy-csharp/pull/82
+[#82]: https://github.com/cronofy/cronofy-csharp/pull/81
```
Comment (gshutler) [inline CHANGELOG.md:275]:
```suggestion
[#82]: https://github.com/cronofy/cronofy-csharp/pull/82
```

**Review by gshutler — CHANGES_REQUESTED** [review]
As this is a major version, we can fix the change 🇬🇧 `DataCentre` to be 🇺🇸 `DataCenter` to match standards too.

Would want to mention it in the changelog, should be an easy fix for people if they are using it, and not in that many spots in their code either.

**Review by gshutler — APPROVED** [review]

**Pull request merged**